reporter/otlp_reporter.go

```diff
@@ -723,15 +727,31 @@ func createFunctionEntry(funcMap map[funcInfo]uint64,
 
 // addProfileAttributes adds attributes to Profile.attribute_table and returns
 // the indices to these attributes.
-func addProfileAttributes(profile *profiles.Profile,
-	attributes []attrKeyValue, attributeMap map[string]uint64) []uint64 {
+func addProfileAttributes[T string | int64](profile *profiles.Profile,
+	attributes []attrKeyValue[T], attributeMap map[string]uint64) []uint64 {
 	indices := make([]uint64, 0, len(attributes))
 
-	addAttr := func(attr attrKeyValue) {
-		if attr.value == "" {
+	addAttr := func(attr attrKeyValue[T]) {
+		var attributeCompositeKey string
+		var attributeValue common.AnyValue
+
+		switch val := any(attr.value).(type) {
+		case string:
+			if val == "" {
+				return
+			}
```

Comment:
How does the receiver of the message distinguish between Key="" and "Key is not available"? I think we should be explicit here and send Key="" (Key with empty string). Same in the int64 case, but for 0 values.

We should not do assumptions here about the receiver's implementation.

```suggestion
```

Reply:
you are right. We should be supporting empty values as specified [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute)